lib/spack/spack/cmd/env.py

```diff
@@ -114,7 +107,7 @@ def env_create(args):
     env = _env_create(
         args.env_name,
         init_file=args.envfile,
-        dir=args.dir,
+        dir=args.dir or "." in args.env_name or "/" in args.env_name,
```

Comment:
(Nitpick) Do we want this:
```console
$ spack env create foo.bar
==> Created environment in: /tmp/foo.bar
==> Activate with: spack env activate foo.bar
```
?

Reply:
You mean test for `"/" in args.env_name` only?

Reply:
Yes, something like that - a dot in the middle of a name is adding an anonymous environment which might be surprising.